src/main.zig

```diff
@@ -2596,6 +2605,23 @@ fn buildOutputType(
             link_libcpp = true;
     }
 
+    const linker_default_entry = linker_force_entry orelse true;
+    if (!linker_default_entry and entry != null) {
+        fatal("combination of '-fentry=[name]' and `-fno-entry` are incompatible", .{});
```

Comment:
Seems like it would be more in line with the other options for `-fno-entry` to set `entry` to `null`.

Reply:
That's a good idea. That will actually solve the issue I was running into which I mentioned in my comment to Andrew. Then when `-fentry` is given, we could overwrite `entry` with a default symbol name if `entry` is `null`.